src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -109,4 +140,67 @@ public function transChoice($id, $number, array $parameters = array(), $domain =
         return (bool) preg_match_all(static::$regexSprintfParams, $string)
             && !(bool) preg_match_all(static::$regexClassicParams, $string);
     }
+
+    /**
+     * Tries to translate the provided message using the legacy system
+     *
+     * @param string $message
+     * @param array $parameters
+     * @param string $domain
+     * @param string|null $locale
+     *
+     * @return mixed|string
+     *
+     * @throws \Exception
+     */
+    private function translateUsingLegacySystem($message, array $parameters, $domain, $locale = null)
+    {
+        $domainParts = explode('.', $domain);
+        if (count($domainParts) < 2) {
+            throw new InvalidArgumentException(sprintf('Invalid domain: "%s"', $domain));
+        }
+
+        $moduleName = strtolower($domainParts[1]);
+        $sourceFile = (!empty($domainParts[2])) ? strtolower($domainParts[2]) : $moduleName;
+
+        // translate using the legacy system WITHOUT fallback to the new system (to avoid infinite loop)
+        return (new LegacyTranslator())->translate($moduleName, $message, $sourceFile, $parameters, false, $locale, false);
+    }
+
+    /**
+     * Indicates if we should try and translate the provided wording using the legacy system.
+     *
+     * @param string $message Message to translate
+     * @param string $domain Translation domain
+     * @param string $translated Message after first translation attempt
+     *
+     * @return bool
+     */
+    private function shouldFallbackToLegacyModuleTranslation($message, $domain, $translated)
+    {
+        return
+            $message === $translated
+            && 'Modules.' === substr($domain, 0, 8)
```

Comment:
current legacy domain names doesn't contains a dot: see https://github.com/PrestaShop/PrestaShop/pull/13059/commits/08b43a35007b7ce314644790c5e190d1343fe138

Reply:
I'm not sure exactly what you're trying to say but modules calling `trans()` should use domains with dots following the schema `Modules.ModuleName<.specific>`

Reply:
this is not the case right now, as legacy files doesn't have any specifics.

Not a big deal, and can be adressed in my PR once the expected behavior will be specified: what should be the value of <specific>?

Reply:
The specific part is [defined in the docs](https://devdocs.prestashop.com/1.7/development/internationalization/translation/translation-domains/#modules) and it's optional. For legacy modules, it's the second parameter to `l()`.

I don't understand why there should be an infinite loop, since a parameter has been added specifically to avoid that 🤔